core/transport/src/main/proto/grpc/block_master.proto

```diff
@@ -223,14 +226,22 @@ message CommitBlockInUfsPRequest {
 message CommitBlockInUfsPOptions {}
 message CommitBlockInUfsPResponse {}
 
-message GetWorkerIdPOptions {}
+message GetWorkerIdPOptions {
+  optional bool extendedRegisterInfo = 1; // Identify whether there is preRegister information
```

Comment:
There's no need for this flag. You can just check `GetWorkerIdRequest.hasClusterId()`

Reply:
done